src/controllers/user/update-user.js

```diff
@@ -0,0 +1,16 @@
+const User = require('../../models/user');
+
+const updateUser = async (username, ...fields) => {
```

Comment:
I would recommend having one endpoint to update the user (when necessary), then another to change the user's password. There is sufficient logic involved with changing a password that could warrant separating this to a new endpoint. Otherwise, you would end up with a very large function that handles all the nuances of changing a password as well as changing other user properties.

Reply:
Good to know! I originally had it as separate functions but thought it was a better idea to have it as a single function as it would reduce the number of functions I have created. But you raised a good point! I will update